torch/onnx/_internal/fx/serialization.py

```diff
@@ -41,11 +46,21 @@ def _create_tensor_proto_with_external_data(
     # FIXME: Avoid importing onnx into torch.onnx.
     import onnx
 
+    scalar_type = jit_type_utils.JitScalarType.from_dtype(
+        _ONNX_TENSOR_ELEMENT_TYPE_TO_TORCH_DTYPE[dtype_override.tensor_type.elem_type]  # type: ignore[index]
+        if dtype_override is not None
+        else tensor.dtype
+    )
+    # Checkpoints can be stored with a different dtype as the model expects because
+    # the user script can explicitly cast the original type to something or maybe
+    # PyTorch's type promotion might do it
+    if dtype_override is not None and dtype_override != tensor.dtype:
```

Comment:
```suggestion
    if dtype_override is not None and scalar_type != tensor.dtype:
```

Reply:
Hmm, not sure if even my suggestion is correct. However it is certain `dtype_override != tensor.dtype` is always True since they are not same type.

Reply:
good catch